### PR TITLE
Sales

### PR DESCRIPTION
**Pull request by raonismaneoto**
captura de tela de 2018-03-05 19-10-32

captura de tela de 2018-03-05 19-11-37

captura de tela de 2018-03-05 19-11-58

### CODE REVIEW

File: src/main/webapp/app/core/main/controllers/createSaleDialogController.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    var app = angular.module("efApp");
+
+    app.controller("CreateSaleDialogController", function CreateSaleDialogController(sales, toastr, 
+        ProductService, SalesService, $uibModalInstance) {
+        var createSaleCtrl = this;
+        createSaleCtrl.products = [];
+        createSaleCtrl.saleProducts = [];
+
+        createSaleCtrl.createSale = function createSale() {
+            var sale = prepareSale();
```
Comment (vsseixaso) [inline src/main/webapp/app/core/main/controllers/createSaleDialogController.js]: var sale = createSaleCtrl.prepareSale();
would not that be correct?

File: src/main/webapp/app/core/main/controllers/createSaleDialogController.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+        createSaleCtrl.cancel = function cancel() {
+            $uibModalInstance.dismiss('cancel');
+        };
+
+        createSaleCtrl.hasProduct = function hasProduct(product) {
+            return _.includes(createSaleCtrl.saleProducts, product);
+        };
+
+        function createSaleInstance() {
```
Comment (vsseixaso) [inline src/main/webapp/app/core/main/controllers/createSaleDialogController.js]: createSaleCtrl.createSaleInstance = function createSaleInstance() {
would not that be correct?

File: src/main/webapp/app/core/main/controllers/createSaleDialogController.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            return new Sale({
+                products: createSaleCtrl.saleProducts,
+                itemsQuantity: createSaleCtrl.itemsQuantity,
+                totalPrice: createSaleCtrl.totalPrice
+            });
+        }
+
+        function getAllProducts() {
```
Comment (vsseixaso) [inline src/main/webapp/app/core/main/controllers/createSaleDialogController.js]: similar

File: src/main/webapp/app/core/main/controllers/createSaleDialogController.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            ProductService.getAllProducts().then(function sucess(response) {
+                createSaleCtrl.products = response.data;
+            });
+        }
+
+        function getTotalPrice() {
```
Comment (vsseixaso) [inline src/main/webapp/app/core/main/controllers/createSaleDialogController.js]: similar

File: src/main/webapp/app/core/main/controllers/createSaleDialogController.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            var totalPrice = 0;
+            _.each(createSaleCtrl.saleProducts, function(product) {
+                totalPrice += product.price;
+            });
+            createSaleCtrl.totalPrice = totalPrice;
+        }
+
+        function prepareSale() {
```
Comment (vsseixaso) [inline src/main/webapp/app/core/main/controllers/createSaleDialogController.js]: similar

File: src/main/webapp/app/core/main/controllers/salesController.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                        return salesCtrl.sales;
+                    },
+                    sale: function() {
+                        return sale;
+                    }
+                }
+            });
+        };
+
+        function getSales() {
```
Comment (vsseixaso) [inline src/main/webapp/app/core/main/controllers/salesController.js]: salesCtrl.getSales = function getSales() {
would not that be correct?

File: src/main/webapp/app/core/main/controllers/createSaleDialogController.js (hunk 1 above)
Comment (raonismaneoto) [reply src/main/webapp/app/core/main/controllers/createSaleDialogController.js]: prepareSale is a private function

File: src/main/webapp/app/core/main/controllers/createSaleDialogController.js (hunk 2 above)
Comment (raonismaneoto) [reply src/main/webapp/app/core/main/controllers/createSaleDialogController.js]: nop, I want that to be an auxiliary function, making it a private function is the best practice to achieve it

**Pull request merged**